compio-buf/src/io_vec_buf.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{
+    t_alloc, IndexedIter, IoBuf, IoBufMut, IoSlice, IoSliceMut, OwnedIterator, SetBufInit,
+};
+
+/// A trait for vectored buffers.
+pub trait IoVectoredBuf: 'static {
+    /// The buffer.
+    type Buf: IoBuf + ?Sized;
+
+    /// The owned iterator that wraps `Self`.
+    type OwnedIter: OwnedIterator<Inner = Self> + IoBuf;
+
+    /// Collected [`IoSlice`]s of the buffers.
+    ///
+    /// # Safety
+    ///
+    /// The return slice will not live longer than self.
+    /// It is static to provide convenience from writing self-referenced
+    /// structure.
+    unsafe fn io_slices(&self) -> Vec<IoSlice> {
+        self.iter_buf().map(|buf| buf.as_io_slice()).collect()
+    }
+
+    /// An iterator over the buffers that's either owned or borrowed with
+    /// [`MaybeOwned`].
```

Comment:
The comment and method seems inconsistent.

Reply:
Right. I need to revert the previous commit